src/test/java/it/eng/idsa/businesslogic/processor/common/RegisterTransactionToCHProcessorTest.java

```diff
@@ -1,32 +1,50 @@
 package it.eng.idsa.businesslogic.processor.common;
 
+import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
+import java.util.Optional;
+
 import org.apache.camel.Exchange;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.mockito.InjectMocks;
 import org.mockito.Mock;
 import org.mockito.MockitoAnnotations;
 import org.springframework.context.ApplicationEventPublisher;
 import org.springframework.test.util.ReflectionTestUtils;
 
+import de.fraunhofer.iais.eis.ContractAgreement;
 import de.fraunhofer.iais.eis.Message;
+import de.fraunhofer.iais.eis.MessageProcessedNotificationMessageBuilder;
 import de.fraunhofer.iais.eis.RejectionReason;
+import it.eng.idsa.businesslogic.configuration.ClearingHouseConfiguration;
 import it.eng.idsa.businesslogic.service.ClearingHouseService;
 import it.eng.idsa.businesslogic.service.RejectionMessageService;
+import it.eng.idsa.businesslogic.usagecontrol.service.UsageControlService;
+import it.eng.idsa.businesslogic.util.Helper;
 import it.eng.idsa.multipart.domain.MultipartMessage;
 import it.eng.idsa.multipart.util.UtilMessageService;
 
 public class RegisterTransactionToCHProcessorTest {
 	
-	@InjectMocks
 	private RegisterTransactionToCHProcessor processor;
 	
 	@Mock
-	private ClearingHouseService clearingHouseService;
+	private ClearingHouseConfiguration configuration;
+	
+	@Mock
```

Comment:
Should those 2 Optionals be mock, or normal objects?

Reply:
Tried out, and you are right it works without mocking them